modelzoo/template.py

```diff
@@ -0,0 +1,174 @@
+import argparse
+import tensorflow as tf
+import os
+
+# Set to INFO for tracking training, default is WARN. ERROR for least messages
+tf.logging.set_verbosity(tf.logging.INFO)
+print("Using TensorFlow version %s" % (tf.__version__))
+
+# Definition of some constants
+CONTINUOUS_COLUMNS = ["I" + str(i) for i in range(1, 14)]  # 1-13 inclusive
+CATEGORICAL_COLUMNS = ["C" + str(i) for i in range(1, 27)]  # 1-26 inclusive
+LABEL_COLUMN = ["clicked"]
+
+
+def add_layer_summary(value, tag):
+    tf.summary.scalar('%s/fraction_of_zero_values' % tag,
+                      tf.nn.zero_fraction(value))
+    tf.summary.histogram('%s/activation' % tag, value)
+
+
+# generate dataset pipline
+def generate_input_data():
+    pass
+
+
+# generate feature columns
+def build_feature_cols():
+    pass
+
+
+class WDL():
+    def __init__(self):
+        self.label = None
+        self.feature = None
+        self.bf16 = False
+        self._is_training = True
+
+        self.input_fn()
+        self.model_fn()
+        self.train_fn()
+        self.metrics_fn()
+
```

Comment:
i don't think we need to call into the function like input_fn, train_fn, etc.

Reply:
'input_fn' will be removed in the class. I want to put the forward process into `model_fn`, and put loss computing, optimizer, metrics including ACC & AUC into a function instead of `train_fu` and `metrics_fn`. What to call this function?
TF's estimator API uses a train_op_fn(loss) to create an optimizer, and the loss computing and metrics are packed into a `Head` instance with other things like computing probability of prediction from logits by sigmod().